### PR TITLE
Defer handling app delegate hooks until app is ready.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/AppDelegate.m [hunk 1]
```diff
@@ -564,33 +566,31 @@ - (void)application:(UIApplication *)application
...
-        UIViewController *fromViewController = [[UIApplication sharedApplication] frontmostViewController];
-        [fromViewController presentViewController:controller
-                                         animated:YES
-                                       completion:^{
-                                           completionHandler(NO);
-                                       }];
-    }
+    }];
+
+    completionHandler(YES);
```
Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: It seems like now we could call the completionHandler twice - always here, and sometimes again after presenting the alertController. Why this change in control flow?

File: Signal/src/AppDelegate.m [hunk 2]
```diff
@@ -630,38 +624,42 @@ - (BOOL)application:(UIApplication *)application
...
-        }
 
-        OutboundCallInitiator *outboundCallInitiator = SignalApp.sharedApp.outboundCallInitiator;
-        OWSAssert(outboundCallInitiator);
-        return [outboundCallInitiator initiateCallWithHandle:phoneNumber];
+            OutboundCallInitiator *outboundCallInitiator = SignalApp.sharedApp.outboundCallInitiator;
+            OWSAssert(outboundCallInitiator);
+            [outboundCallInitiator initiateCallWithHandle:phoneNumber];
+        }];
+        return YES;
```
Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: I'm unfamiliar with the ramifications of always returning YES here.

File: Signal/src/AppDelegate.m (hunk 1 above)
Comment (charlesmchen) [reply Signal/src/AppDelegate.m]: Good catch, this belongs inside the block above.

File: Signal/src/AppDelegate.m (hunk 2 above)
Comment (charlesmchen) [reply Signal/src/AppDelegate.m]: I've added a comment.
